R/LearnerRegrKSVM.R

```diff
@@ -0,0 +1,80 @@
+        ParamFct$new(id = "kernel", default = "rbfdot", levels = c("rbfdot","polydot","vanilladot","tanhdot","laplacedot","besseldot","anovadot","splinedot","stringdot"), tags = c("train")),
+        ParamDbl$new(id = "C", default = 1, tags = c("train")),
+        ParamDbl$new(id = "nu", default = 0.2, tags = c("train")),
+        ParamDbl$new(id = "epsilon", default = 0.1, tags = c("train")),
+        ParamDbl$new(id = "cache", default = 40, tags = c("train")),
+        ParamDbl$new(id = "tol", default = 0.001, tags = c("train")),
+        ParamLgl$new(id = "shrinking", default = TRUE, tags = c("train")),
+
+        # kernel hyperparameters
+        ParamDbl$new(id = "sigma", default = NULL, tags = c("predict", "kpar"), special_vals = list(NULL)),
+        ParamInt$new(id = "degree", default = NULL, tags = c("predict", "kpar"), special_vals = list(NULL)),
+        ParamDbl$new(id = "scale", default = NULL, tags = c("predict", "kpar"), special_vals = list(NULL)),
+        ParamInt$new(id = "order", default = NULL, tags = c("predict", "kpar"), special_vals = list(NULL)),
+        ParamDbl$new(id = "offset", default = NULL, tags = c("predict", "kpar"), special_vals = list(NULL)),
+        ParamInt$new(id = "length", default = NULL, tags = c("predict", "kpar"), special_vals = list(NULL)),
+        ParamDbl$new(id = "lambda", default = NULL, tags = c("predict", "kpar"), special_vals = list(NULL)),
+        ParamLgl$new(id = "normalized", default = NULL, tags = c("predict", "kpar"), special_vals = list(NULL))
+      ))
+
+      ps$add_dep("sigma", "kernel", CondAnyOf$new(c("rbfdot", "laplacedot", "besseldot", "anovadot")))
+      ps$add_dep("degree", "kernel", CondAnyOf$new(c("polydot", "anovadot")))
+      ps$add_dep("scale", "kernel", CondAnyOf$new(c("polydot", "tanhdot")))
+      ps$add_dep("order", "kernel", "besseldot")
+      ps$add_dep("offset", "kernel", CondAnyOf$new(c("polydot", "tanhdot")))
+      ps$add_dep("length", "kernel", "stringdot")
+      ps$add_dep("lambda", "kernel", "stringdot")
+      ps$add_dep("normalized", "kernel", "stringdot")
+
+      super$initialize(
+        id = id,
+        packages = "kernlab",
+        feature_types = c("logical", "integer", "numeric", "character", "factor", "ordered"),
+        predict_types = c("response"),
+        param_set = ps,
+        properties = c("weights")
+      )
+    },
+
+    train_internal = function(task) {
+      pars = self$param_set$get_values(tags = "train")
+      kpar = self$param_set$get_values(tags = "kpar")
+      if(length(kpar) == 0) {
+        pars$values = list(kpar = "automatic")
+      } else {
+        pars$values = list(kpar = kpar)
+      }
+
+      f = task$formula()
+      data = task$data()
+      invoke(kernlab::ksvm, x = f, data = data, class.weights = task$weights$weight, .args = pars)
+    },
+
+    predict_internal = function(task) {
+      newdata = task$data(cols = task$feature_names) #get newdata
```

Comment:
You might want to add `newdata = newdata[, self$state$feature_names, drop = FALSE]` as present in "LearnerSVM" in mlr3learners. I am not sure how important that line is but staying consistent with mlr3learners is certainly not bad.